R/MixedModelsCommon.R

```diff
@@ -792,1293 +793,4580 @@
+        se         = tempSummary$MAD_SD[j],
+        lowerCI    = tempSummary[j, paste0((.50 - options$summaryCI / 2) *
+                                             100, "%")],
+        upperCI    = tempSummary[j, paste0((.50 + options$summaryCI / 2) *
+                                             100, "%")],
+        rhat       = tempSummary$rhat[j],
+        ess_bulk   = tempSummary$ess_bulk[j],
+        ess_tail   = tempSummary$ess_tail[j]
+      )
+
+      if (varName != "Intercept" && nrow(tempSummary) > 1) {
+        varName <-
+          paste(.unv(unlist(strsplit(
+            as.character(tempSummary$Variable[j]), ","
+          ))), collapse = jaspBase::interactionSymbol)
+        varName <- gsub(" ", "", varName, fixed = TRUE)
+        if (grepl(jaspBase::interactionSymbol, names(modelSummary)[i], fixed = T)) {
+          for (n in unlist(strsplit(.unv(names(
+            modelSummary
+          )[i]), jaspBase::interactionSymbol))) {
+            varName <- gsub(n, "", varName, fixed = TRUE)
+          }
+        } else{
+          varName <-
+            gsub(.unv(names(modelSummary)[i]), "", varName, fixed = TRUE)
+        }
+        tempRow$level <- varName
+      }
+
+      tempTable$addRows(tempRow)
+    }
+
+    # add message about (lack of) random effects grouping factors
+    tempTable$addFootnote(.mmMessageREgrouping(options$randomVariables))
+
+    # check model fit
+    divIterations <- rstan::get_num_divergent(model$stanfit)
+    lowBmfi       <- rstan::get_low_bfmi_chains(model$stanfit)
+    maxTreedepth  <- rstan::get_num_max_treedepth(model$stanfit)
+    if(any(is.infinite(rstan::summary(model$stanfit)$summary[, "Rhat"]))){
+      maxRhat     <- Inf
+    }else{
+      maxRhat     <- max(rstan::summary(model$stanfit)$summary[, "Rhat"])
+    }
+    minESS        <-
+      min(rstan::summary(model$stanfit)$summary[, "n_eff"])
+    if (divIterations != 0) {
+      tempTable$addFootnote(.mmMessageDivergentIter(divIterations), symbol = gettext("Warning:"))
+    }
+    if (length(lowBmfi) != 0) {
+      tempTable$addFootnote(.mmMessageLowBMFI(length(lowBmfi)), symbol = gettext("Warning:"))
+    }
+    if (maxTreedepth != 0) {
+      tempTable$addFootnote(.mmMessageMaxTreedepth(max_treedepth))
+    }
+    if (maxRhat > 1.01) {
+      tempTable$addFootnote(.mmMessageMaxRhat(maxRhat), symbol = gettext("Warning:"))
+    }
+    if (minESS < 100 * options$chains || is.nan(minESS)) {
+      tempTable$addFootnote(.mmMessageMinESS(minESS, 100 * options$chains), symbol = gettext("Warning:"))
+    }
+
+    removedMe <- jaspResults[["mmModel"]]$object$removedMe
+    removedTe <- jaspResults[["mmModel"]]$object$removedTe
+    addedRe   <- jaspResults[["mmModel"]]$object$addedRe
+    if (length(removedMe) > 0) {
+      for (j in 1:length(removedMe)) {
+        tempTable$addFootnote(.mmMessageOmmitedTerms1(removedMe[[j]], names(removedMe)[j]),
+                              symbol = gettext("Note:"))
+      }
+    }
```

Comment:
```suggestion
    for (j in seq_along(removedMe))
      tempTable$addFootnote(.mmMessageOmmitedTerms1(removedMe[[j]], names(removedMe)[j]),
                              symbol = gettext("Note:"))
```
The same goes for the 2 other if statements and loop below this one. In general, `seq_along(x)` is *always* preferred over `1:length(x)`, precisely so you never need this additional if statements.

Reply:
fixed